diff --git a/src/CascadeMinimizer.cc b/src/CascadeMinimizer.cc
index 5090eb7..7cf470a 100644
--- a/src/CascadeMinimizer.cc
+++ b/src/CascadeMinimizer.cc
@@ -77,6 +77,7 @@ bool CascadeMinimizer::improveOnce(int verbose)
     static int rooFitOffset = runtimedef::get("MINIMIZER_rooFitOffset");
     std::string myType(ROOT::Math::MinimizerOptions::DefaultMinimizerType());
     std::string myAlgo(ROOT::Math::MinimizerOptions::DefaultMinimizerAlgo());
+    minimizer_->setEps(ROOT::Math::MinimizerOptions::DefaultTolerance());
     bool outcome = false;
     if (oldFallback_){
         if (optConst) minimizer_->optimizeConst(std::max(0,optConst));
@@ -192,7 +193,10 @@ bool CascadeMinimizer::minimize(int verbose, bool cascade)
 
     minimizer_->setPrintLevel(verbose-2);  
     minimizer_->setStrategy(strategy_);
-    if (preScan_) minimizer_->minimize("Minuit2","Scan");
+    if (preScan_) {
+      minimizer_->setEps(ROOT::Math::MinimizerOptions::DefaultTolerance());
+      minimizer_->minimize("Minuit2","Scan");
+    }
 
     
     //if (preFit_ && nuisances != 0) {
